src/com/perl5/lang/perl/idea/run/PerlRunProfileState.java

```diff
@@ -155,4 +163,54 @@ protected ProcessHandler startProcess() throws ExecutionException
 		return runProfile.getEnvs();
 	}
 
+	public List<String> getPerlLibraries(@Nullable VirtualFile scriptFile)
+	{
+		if (!PlatformUtils.isIntelliJ())
+		{
+			return Collections.emptyList();
+		}
+		else
+		{
+			PerlConfiguration runProfile = (PerlConfiguration) getEnvironment().getRunProfile();
+
+			// try to get used SDK
+			Sdk sdk;
+			if (runProfile.isUseAlternativeSdk())
+			{
+				sdk = ProjectJdkTable.getInstance().findJdk(runProfile.getAlternativeSdkPath());
+			}
+			else
+			{
+				Module moduleForFile = scriptFile == null ? null : ModuleUtilCore.findModuleForFile(scriptFile, getEnvironment().getProject());
+
+				sdk = PerlRunUtil.getModuleSdk(moduleForFile);
+				if (sdk == null)
+				{
+					// try project SDK
+					sdk = ProjectRootManager.getInstance(getEnvironment().getProject()).getProjectSdk();
+				}
+			}
+
+			if (sdk != null && sdk.getSdkType() == PerlSdkType.getInstance())
+			{
+				List<String> incPaths = PerlSdkType.getInstance().getINCPaths(sdk.getHomePath());
+				List<String> list = new ArrayList<>();
+
+				for (VirtualFile file : sdk.getRootProvider().getFiles(OrderRootType.CLASSES))
```

Comment:
It looks like you are missing whole picture:

1. There is an SDK (perl interpreter in our case). It has classroots - `@INC`. SDK is application-wide. same for all projects.
2. There is a project. 
3. There are some number of modules in the project and each can have it's libraries too - added by user as module dependencies or by marking some directories as libraries in project structure (we have no external libraries interface for Perl modules for now, but this is an IDEA architecture). 
4. Module may depend on other modules and use their dependencies (we have no interface for it in Perl module, but this is an IDEA architecture).
5. One SDK may be assigned to the project.
5. One SDK may be assigned to a module (or module may inherits SDK from it's project)

Generally if we have a file in some module, classroots for it are built the following way:
`Classroots from module + Classroots exported by dependent modules + classroots from module or project SDK`

In our case, if we need to run external process with our file classroots using module(project) SDK or alternative sdk we need all of them except classroots from SDK, because perl already knows about `@INC`

There is no sense to properly collect all classroots according to the logic above. LibraryUtil does this for you.

One thing you should do manualy I guess - remove libroots obtained from shared settings. This is a setback of my old bad architecture :)

After explaining all of this, I figured out that the only way you could add dependencies to the perl module - You've added it to an SDK. This is a wrong way.

Try to enable classpatheditor in `com.perl5.lang.perl.idea.configuration.module.PerlModuleConfigurationEditorProvider`, add your dependencies in module dependencies and use LibraryUtil.